main/src/cgeo/geocaching/utils/CalculationUtils.java

```diff
@@ -0,0 +1,128 @@
+package cgeo.geocaching.utils;
+
+public final class CalculationUtils {
+    private CalculationUtils() {
+        // utility class
+    }
+
+    /**
+     * This simple evaluation algorithm was created by user 'Boann' and released to the public domain on Stack Overflow:
+     * https://stackoverflow.com/questions/3422673/evaluating-a-math-expression-given-in-string-form
+     *
+     * I added the '%' operator myself.
+     * Note that the 'sqrt', 'cos', 'sin' and 'tan' functions don't work in this implementation as
+     * expected as the individual letters are automatically replaced with their corresponding
+     * expressions as they are interpreted as being individual variables.
+     **/
+    public static double eval(final String str) {
+        return new Object() {
```

Comment:
Or just refactor this class from a utility class `CalculationUtil` to an instantiate-able class `Calculator`, put the expression `str` into the constructor and store it in a field. Then have `eval()` without argument and extract all the methods from the anonymous class into the outer class `Calculator`.